docs/adrs/00005-license-export-analytics.md

```diff
@@ -0,0 +1,114 @@
+#### 2 Package-level licenses
+##### 2.1 CycloneDX
+Contains three types: Id, name, and Expression
+The Expression follows the SPDX expression standard.
+```json
+"licenses" : [  {    "license" : {      "id" : "EPL-1.0"    }  },  {    "license" : {      "name" : "GNU Lesser General Public License",      "url" : "http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html"    }  }],`
+
+```
+##### 2.2 SPDX
+It's Expression and follows the SPDX expression standard.
+```json
+"licenseDeclared": "Apache-2.0 OR LicenseRef-GPL-2.0-with-classpath-exception",
+```
+##### 2.3 Special Types in SPDX: ExtractedLicensingInfos
+```json
+"hasExtractedLicensingInfos": [
+{
+"comment": "External License Info is obtained from a build system which predates the SPDX specification and is not strict in accepting valid SPDX licenses.",
+"extractedText": "The license info found in the package meta data is: GPL-2.0-with-classpath-exception. See the specific package info in this SPDX document or the package itself for more details.",
+"licenseId": "LicenseRef-GPL-2.0-with-classpath-exception",
+"name": "GPL-2.0-with-classpath-exception"
+},
+```
+The licenseId can be associated with the license in the package.
+### 2 Current license export requirements
+https://docs.google.com/spreadsheets/d/1Ki9hkLl94L3G4A4rfQQ4wuEnQFTPkpNflClVjlk2mQA/edit?gid=0#gid=0
+
+### 3 Current license data structure design in trustify
+
+```rust
+pub struct Model {
+    #[sea_orm(primary_key)]
+    pub id: Uuid,
+    pub text: String,
+    pub spdx_licenses: Option&lt;Vec&lt;String&gt;&gt;,
+    pub spdx_license_exceptions: Option&lt;Vec&lt;String&gt;&gt;,
+}
+```
+The text field corresponds to CycloneDX's Id, Name, or complete Expression. When it's an Expression, it will be parsed into a license array and stored in spdx_licenses, while parsing exceptions are stored in spdx_license_exceptions.
+For SPDX, the text field corresponds to the complete Expression. Similarly, the Expression will be parsed into a license array and stored in spdx_licenses, with parsing exceptions stored in spdx_license_exceptions.
+Currently missing SPDX's ExtractedLicensingInfos and SBOM-level licenses.
+
+### 4 License data structure in GUAC
+![erd_guac.png](erd_guac.png)
+GUAC parses all expressions into individual licenses and saves them separately. LicenseRef* represents ExtractedLicensingInfos, with its extractedText stored in the inline field.
+
+## Decision
+
+### Current design proposal
+![erd_license.png](erd_license.png)
+```rust
+pub enum LicenseCategory {
+    #[sea_orm(string_value = "slc")]
+    SPDXDECLARED,
+    #[sea_orm(string_value = "sld")]
+    SPDXCONCLUDED,
+    #[sea_orm(string_value = "clci")]
+    CYDXLCID,
+    #[sea_orm(string_value = "clcn")]
+    CYDXLCNAME,
+    #[sea_orm(string_value = "cle")]
+    CYDXLEXPRESSION,
+    #[sea_orm(string_value = "cd")]
+    CLEARLYDEFINED,
+    #[sea_orm(string_value = "o")]
+    OTHER,
+}
+```
+SPDXDECLARED: The license from spdx sbom's declered license.
```

Comment:
I don't understand why we'd need different types here. Why do we differentiate between CycloneDX and SPDX? Why would we attach that "type" information to the table hold general license information?

Reply:
Currently, the requirement is only for the declared licenses in SPDX and the licenses in CycloneDX. So it is important to know where it comes from, because in SPDX there is also a concluded license, and some of these licenses are derived from clearly defined sources.

Reply:
In the database we do not store the type of the document (SPDX or CDX). Why is that important to know? From what I see it, the information is attached to the `license` table. Which doesn't have any SBOM information.

Reply:
agree with @ctron - a license (defined or concluded)  is independent to serialisation format eg. does not matter if it was serialised in CDX or SPDX (or what capabilities to represent in either) ...